### PR TITLE
fix: [CDS-41789]: added runtime API support for env v2 for sshWinRm Azure

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

added runtime API support for env v2 for sshWinRm Azure

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10585)

**Review by YounixIO — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWimRmAzureInfrastructureSpecInputForm.tsx [hunk 1]
```diff
@@ -172,43 +205,40 @@ const SshWinRmAzureInfrastructureSpecInputFormNew: React.FC<AzureInfrastructureS
...
+    useEffect(() => {
+      const options =
+        resourceGroupDataV2?.data?.resourceGroups?.map(rg => ({ label: rg.resourceGroup, value: rg.resourceGroup })) ||
+        /* istanbul ignore next */ []
+      setResourceGroups(options)
+    }, [resourceGroupDataV2])
+
+    const fetchResourceUsingEnvId = (): boolean => {
+      return (
+        getMultiTypeFromValue(connectorRef) !== MultiTypeInputType.RUNTIME &&
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWimRmAzureInfrastructureSpecInputForm.tsx:232]: Add comments on the logic around this function

**Comment by vikrant-harness** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWimRmAzureInfrastructureSpecInputForm.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWimRmAzureInfrastructureSpecInputForm.tsx:232]: @YounixIO added this in the latest commit

**Review by Eric761 — DISMISSED** [review]

**Review by Sagar-Rajput7 — DISMISSED** [review]

**Review by YounixIO — APPROVED** [review]

**Comment by vikrant-harness** [comment]
retrigger jest

**Comment by vikrant-harness** [comment]
retrigger coverage

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWimRmAzureInfrastructureSpecInputForm.tsx [hunk 2]
```diff
@@ -174,43 +207,41 @@ const SshWinRmAzureInfrastructureSpecInputFormNew: React.FC<AzureInfrastructureS
...
 
     useEffect(() => {
       const options =
         resourceGroupData?.data?.resourceGroups?.map(rg => ({ label: rg.resourceGroup, value: rg.resourceGroup })) ||
         /* istanbul ignore next */ []
       setResourceGroups(options)
     }, [resourceGroupData])
+    useEffect(() => {
+      const options =
+        resourceGroupDataV2?.data?.resourceGroups?.map(rg => ({ label: rg.resourceGroup, value: rg.resourceGroup })) ||
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWimRmAzureInfrastructureSpecInputForm.tsx:227]: nit: we can do something similar to line 212

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWinRmAzureInfrastructureForm.tsx [hunk 3]
```diff
@@ -342,10 +342,13 @@ export const AzureInfrastructureSpecForm: React.FC<AzureInfrastructureSpecEditab
                           formik.setFieldValue('connectorRef', connectorRef)
                         }
                         getMultiTypeFromValue(formik.values?.subscriptionId) !== MultiTypeInputType.RUNTIME &&
+                          formik.values?.subscriptionId &&
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWinRmAzureInfrastructureForm.tsx:345]: I don't see a useValue on the form input for this. Is formik.values?.subscriptionId not giving an empty object instead?

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by kapilkumawat86 — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWimRmAzureInfrastructureSpecInputForm.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWimRmAzureInfrastructureSpecInputForm.tsx]: addressed in the latest commit

File: src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWinRmAzureInfrastructureForm.tsx (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/SshWinRmAzureInfrastructureSpec/SshWinRmAzureInfrastructureForm.tsx]: addressed in the latest commit

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Comment by vikrant-harness** [comment]
retrigger jest

**Comment by vikrant-harness** [comment]
retrigger coverage

**Comment by vikrant-harness** [comment]
retrigger jest

**Comment by vikrant-harness** [comment]
retrigger coverage

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**